### PR TITLE
Implement editors/devfile route

### PR DESCRIPTION
**Pull request by dkwon17**
### What does this PR do?
This PR adds a new route exposed by the dashboard backend:
image

### What issues does this PR fix or reference?
https://github.com/eclipse-che/che/issues/22994

### Is it tested? How?

When testing on Eclipse Che, access the following link and verify that the `che-code/latest` devfile is returned:
```
<CHE-HOST>/dashboard/api/editors/devfile?che-editor=che-incubator/che-code/latest
```

If no `che-editor` query param is provided, then there is a 400 error:
```
<CHE-HOST>/dashboard/api/editors/devfile?che-editor=che-incubator/che-code/latest
```

If there is not editor with id `che-editor`, there is a 404 error:
```
<CHE-HOST>/dashboard/api/editors/devfile?che-editor=doesnotexist
```

Additionally, create this following devworkspace and verify that the workspace starts and verify that you can access the editor:
```
kind: DevWorkspace
apiVersion: workspace.devfile.io/v1alpha2
metadata:
  name: my-devworkspace
spec:
  routingClass: che
  started: true
  contributions:
    - name: ide
      uri: http://che-dashboard.eclipse-che.svc.cluster.local:8080/dashboard/api/editors/devfile?che-editor=che-incubator/che-code/latest
  template:
    projects:
      - name: my-project-name
        git:
          remotes:
            origin: https://github.com/eclipse-che/che-docs
    components:
      - name: tooling-container
        container:
          image: quay.io/devfile/universal-developer-image:ubi8-latest
```

### CODE REVIEW

**Review by ibuziuk — COMMENTED** [review]
@dkwon17 great job. Are the custom editors going to be exposed as well if configured ?
https://github.com/eclipse-che/che-docs/pull/2738

**Comment by dkwon17** [comment]
@ibuziuk thank you, yes, custom editors are also exposed and are accessible from the new endpoint

**Review by ibuziuk — APPROVED** [review]
LGTM, but it seems there is a linting error that should be fixed via `yarn lint:fix`

**Review by olexii4 — APPROVED** [review]

File: packages/dashboard-frontend/src/contexts/WorkspaceActions/BulkDeleteButton/__mocks__/index.tsx [hunk 1]
```diff
@@ -21,8 +21,8 @@ export class WorkspaceActionsBulkDeleteButton extends React.Component<Props> {
     const { isDisabled, onAction, workspaces } = this.props;
 
     const handleAction = () =>
-      workspaces.map(workspace =>
-        onAction?.(WorkspaceAction.DELETE_WORKSPACE, workspace.uid, true),
+      workspaces.map(
+        workspace => onAction?.(WorkspaceAction.DELETE_WORKSPACE, workspace.uid, true),
```
[the remark this answers is not in the document]
Comment (dkwon17) [reply packages/dashboard-frontend/src/contexts/WorkspaceActions/BulkDeleteButton/__mocks__/index.tsx]: Fixed, but for some reason there is a formatting error on the dogfooding cluster:
Screenshot from 2024-06-20 18-51-08

File: packages/dashboard-backend/src/routes/api/helpers/__mocks__/getDevWorkspaceClient.ts [hunk 2]
```diff
@@ -140,89 +154,90 @@ export const stubAutoProvision = true;
...
-      patch: (_namespace, _gitconfig) => Promise.resolve({} as api.IGitConfig),
-    } as IGitConfigApi,
-    sshKeysApi: {
-      add: (_namespace, _sshKey) => Promise.resolve({} as api.SshKey),
-      delete: (_namespace, _name) => Promise.resolve(),
-      list: _namespace => Promise.resolve(stubSshKeysList),
-    } as IShhKeysApi,
-  } as DevWorkspaceClient;
-}
+export const getDevWorkspaceClient = jest.fn(
```
Comment (dkwon17) [inline packages/dashboard-backend/src/routes/api/helpers/__mocks__/getDevWorkspaceClient.ts:157]: This was done in order to provide certain mock implementations for certain tests:
https://github.com/eclipse-che/che-dashboard/blob/64cb29d42acfc59104aac5ffd76ba5d221b56be0/packages/dashboard-backend/src/routes/api/__tests__/editors.spec.ts#L67

File: packages/dashboard-backend/src/routes/api/editors.ts [hunk 3]
```diff
@@ -24,7 +26,34 @@ export function registerEditorsRoutes(instance: FastifyInstance) {
...
+        try {
+          editor = await editorsApi.get(editorId);
+        } catch (error) {
+          if (error instanceof EditorNotFoundError) {
+            return reply.status(404).send(error);
+          }
+          throw error;
+        }
+
+        return dump(editor);
```
Comment (dkwon17) [inline packages/dashboard-backend/src/routes/api/editors.ts:55]: I was able to test this code without creating a new helper function

Is it ok @olexii4?

**Comment by ibuziuk** [comment]
@dkwon17 please, also contribute to the docs - https://eclipse.dev/che/docs/next/administration-guide/configuring-editors-definitions/

**Comment by dkwon17** [comment]
@ibuziuk here is the docs PR: https://github.com/eclipse-che/che-docs/pull/2747

File: packages/dashboard-backend/src/routes/api/editors.ts (hunk 3 above)
Comment (olexii4) [reply packages/dashboard-backend/src/routes/api/editors.ts:55]: It is OK. Thanks.

**Review by olexii4 — APPROVED** [review]

**Comment by ibuziuk** [comment]
@dkwon17 I believe we are good to merge together withhttps://github.com/eclipse-che/che-docs/pull/2747

**Pull request merged**